### ISSUE TITLE
dynamic import at top level fails

### ISSUE THREAD

**Issue by s7dhansh** [ardatan/meteor-webpack#64]
A)
```
async function main() {
	const [
		React,
		{hydrate},
	] = await Promise.all([
		import('react'),
		import('react-dom'),
	]);

	hydrate(<div>asdfasdf</div>, document.getElementById('root'));
}

main();
```

fails with an error:
```
Uncaught ReferenceError: exports is not defined
    at 0.js:1
(anonymous) @ 0.js:1
app.js:817 Uncaught (in promise) Error: Loading chunk 0 failed.
```

If I remove the Promises, it works:
B)
```
import React from 'react';
import {hydrate} from 'react-dom';
hydrate(<div>asdfasdf</div>, document.getElementById('root'));
```

Ironically enough, if I replace (B) with (A) now (while meteor is running), the app starts working. It fails again when I restart meteor. Can someone guide me on fixing/debugging this?

**Comment by abecks** [comment]
I don't believe Meteor's dynamic import functionality will currently work with this package. Some more work is likely required. You might be better off using webpack's code splitting functionality.

**Comment by s7dhansh** [comment]
How do I do that?
Webpack's code splitting with dynamic imports also uses the same syntax, right? Should I remove dynamic-import from meteor packages?

**Comment by abecks** [comment]
To be honest I'm not sure, as I haven't used it before. If you have any
luck, please let me know.

**Comment by ardatan** [comment]
@s7dhansh @abecks Meteor-Webpack supports dynamic imports by serving chunks as assets;
https://github.com/ardatan/meteor-webpack/blob/1f6796743d226c6fb65e5cf4f5c16e24ef3c3c71/atmosphere-packages/webpack/plugin.js#L129

I think there is a problem in dev-middleware package.

**Comment by s7dhansh** [comment]
@ardatan sorry I have very limited understanding of what you are saying. Can you let me know a fix that I can try?
